src/implementations/config/create.ts

```diff
@@ -0,0 +1,211 @@
+      fund: "ropsten",
+    };
+    walletPath = await createWallet(createWalletParams);
+    info(`Wallet created at ${walletPath}`);
+  }
+  const walletFilePath = walletPath || encryptedWalletPath;
+  const wallet = await readFile(walletFilePath);
+  const walletObject = JSON.parse(wallet);
+
+  info(`Wallet detected at ${walletFilePath}`);
+
+  const configFile: ConfigFile = await selectTemplateFile(configType, configTemplatePath);
+  if (configFile.wallet.type === "ENCRYPTED_JSON") {
+    configFile.wallet.encryptedJson = wallet;
+  }
+
+  const formsInTemplate = configFile.forms;
+
+  const createDocumentStore = async (): Promise<string> => {
+    info(`Enter password to continue deployment of Document Store`);
+    const deployDocumentStoreParams = {
+      encryptedWalletPath: walletFilePath,
+      network: "ropsten",
+      gasPriceScale: 1,
+      dryRun: false,
+      storeName: "Document Store",
+    };
+    const { contractAddress } = await deployDocumentStore(deployDocumentStoreParams);
+    success(`Document store deployed, address: ${highlight(contractAddress)}`);
+    return contractAddress;
+  };
+
+  const createTokenRegistry = async (): Promise<string> => {
+    info(`Enter password to continue deployment of Token Registry`);
+    const deployTokenRegistryParams = {
+      registryName: "Demo Token Registry",
+      registrySymbol: "DTR",
+      encryptedWalletPath: walletFilePath,
+      network: "ropsten",
+      gasPriceScale: 1,
+      dryRun: false,
+    };
+    const { contractAddress } = await deployTokenRegistry(deployTokenRegistryParams);
+    success(`Token registry deployed, address: ${highlight(contractAddress)}`);
+    return contractAddress;
+  };
+
+  const generateDocumentStoreOrTokenRegistryOrDid = async (
+    typesOfForms: TypesOfForms[],
+    formType: "VERIFIABLE_DOCUMENT" | "TRANSFERABLE_RECORD",
+    didOrTxt: "DNS-DID" | "DNS-TXT"
+  ): Promise<string> => {
+    const shouldGenerate =
+      typesOfForms.filter(
+        (item: TypesOfForms) =>
+          item.type === formType && (<any>Object).values(item.identityProofTypes).includes(didOrTxt)
+      ).length > 0;
+
+    if (!shouldGenerate) return "";
+    if (formType === "VERIFIABLE_DOCUMENT" && didOrTxt === "DNS-TXT") return await createDocumentStore();
+    if (formType === "VERIFIABLE_DOCUMENT" && didOrTxt === "DNS-DID") {
+      info(`Creating temporary DNS for DID`);
+      return (
+        (await createTemporaryDns({
+          networkId: 3,
+          publicKey: `did:ethr:0x${walletObject.address}#controller`,
+          sandboxEndpoint: sandboxEndpointUrl,
+        })) || ""
+      );
+    }
+    if (formType === "TRANSFERABLE_RECORD") return await createTokenRegistry();
+    return "";
+  };
+
+  // loop through the form template to check the type of forms
+  const typesOfForms: TypesOfForms[] = formsInTemplate.map((form) => {
+    const identityProofTypes = form.defaults.issuers.map((issuer: Issuer) => issuer.identityProof?.type);
```

Comment:
134 to 194 should move out also to `getUpdatedConfigFile` or something

Reply:
im thinking along the lines:

```
const updatedForms = forms.map(() => ...); // do what is needed
const updatedConfigFile = { ...configFile, forms: updatedForms };
```

Reply:
ok, i think can refactor abit